File: thoughts/2026-08-28-search-rust-rewrite/teammates/session-core/probes/trim_probe.py
from chats.parsing import decode_jsonl_entries, detect_format

CASES = [
    ("U+001C FS", ""),
    ("U+001F US", ""),
    ("U+0085 NEL", ""),
    ("U+00A0 NBSP", " "),
    ("U+2028 LS", " "),
    ("U+3000 IDSP", "　"),
    ("U+200B ZWSP", "​"),
]

for name, prefix in CASES:
    content = prefix + '{"type":"user","message":{"role":"user","content":"hi"}}'
    print(
        f"{name:13} py_isspace={str(prefix.isspace()):5} "
        f"decoded={len(decode_jsonl_entries(content))} detect={detect_format(content)}"
    )

File: thoughts/2026-08-28-search-rust-rewrite/teammates/session-core/probes/trim_probe.rs
fn main() {
    let cases = [
        ("U+001C FS", '\u{1c}'),
        ("U+001F US", '\u{1f}'),
        ("U+0085 NEL", '\u{85}'),
        ("U+00A0 NBSP", '\u{a0}'),
        ("U+2028 LS", '\u{2028}'),
        ("U+3000 IDSP", '\u{3000}'),
        ("U+200B ZWSP", '\u{200b}'),
    ];
    for (name, character) in cases {
        let line = format!("{character}{{\"type\":\"user\"}}");
        println!(
            "{:13} rust_trim_strips={}",
            name,
            line.trim().starts_with('{')
        );
    }
}
